clippy_lints/src/utils/conf.rs

```diff
@@ -205,10 +205,33 @@ macro_rules! define_Conf {
 }
 
 define_Conf! {
-    /// Lint: Arithmetic.
+    /// Lint: arithmetic_side_effects.
     ///
-    /// Suppress checking of the passed type names.
-    (arithmetic_side_effects_allowed: rustc_data_structures::fx::FxHashSet<String> = <_>::default()),
```

Comment:
I regret not categorizing this lint as `nursery` in the first place. As you saw yourself, many things changed, bugs were fixed and probably other things will come into play in the the future.

This lint already broke downstream users in stable toolchains because of https://github.com/rust-lang/rust-clippy/pull/9443 so can you guys allow the introduction of these new parameters without having to keep `arithmetic_side_effects_allowed`? For what is worth, there is only one project relying on the current behavior in all of GitHub.

Reply:
The changes would get into I think 1.67 or 1.68 meaning there will be a good few months where the lint is available with the current config on stable

My preference would be to keep the existing variable around, maybe renamed with `conf_deprecated` to `arithmetic_side_effects_allowed_any`

Reply:
Well, at least I tried. Done as requested.